src/Sentry/Integrations/WinUIUnhandledExceptionIntegration.cs

```diff
@@ -0,0 +1,133 @@
+#if NET5_0_OR_GREATER 
+using System;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace Sentry.Integrations
+{
+    // This integration hooks unhandled exceptions in WinUI 3.
+    // The primary hook is Microsoft.UI.Xaml.Application.Current.UnhandledException
+    //
+    // There are some quirks to be aware of:
+    //
+    // - By default, important details (message, stack trace, etc.) are stripped away.
+    //   We can work around this by catching first-chance exceptions.
+    //   See: https://github.com/microsoft/microsoft-ui-xaml/issues/7160
+    //
+    // - Exceptions from background threads are not caught here.
+    //   However, they are caught by System.AppDomain.CurrentDomain.UnhandledException,
+    //   which we already hook in our AppDomainUnhandledExceptionIntegration
+    //   See: https://github.com/microsoft/microsoft-ui-xaml/issues/5221
+    //
+    // Note that we use reflection in this integration to get at WinUI code.
+    // If we ever add a Windows platform target (net6.0-windows, etc.), we could refactor to avoid reflection.
+    //
+    // This integration is for WinUI 3.  It does NOT work for UWP (WinUI 2).
+    // For UWP, the calling application will need to hook the event handler.
+    // See https://docs.sentry.io/platforms/dotnet/guides/uwp/
+    // (We can't do it automatically without a separate UWP class library,
+    // due to a security exception when attempting to attach the event dynamically.)
+
+    internal class WinUIUnhandledExceptionIntegration : ISdkIntegration
+    {
+        private static readonly Assembly? WinUIAssembly = GetWinUIAssembly();
+
+        private Exception? _lastFirstChanceException;
+        private IHub _hub = null!;
+        private SentryOptions _options = null!;
+
+        // Constants used for reflection
+        private const string WinUIAssemblyName = "Microsoft.WinUI";
+        private const string WinUiAssemblyPublicKeyToken = "de31ebe4ad15742b";
+        private const string WinUINamespace = "Microsoft.UI.Xaml";
+        private const string ApplicationTypeName = "Application";
+        private const string UnhandledExceptionEventName = "UnhandledException";
+        private const string CurrentPropertyName = "Current";
+        private const string HandledPropertyName = "Handled";
```

Comment:
have to say i have never been a fan of single use constants disconnected from the context of the use. 

when reading

```
 var handled = (bool)eventArgsType.GetProperty(HandledPropertyName)!.GetValue(e)!;
```

the constant is just more obscured (and more verbose) compared to 

```
 var handled = (bool)eventArgsType.GetProperty("Handled")!.GetValue(e)!;
```